### ISSUE TITLE
Running Glusto Tests Upstream

### ISSUE THREAD

**Issue by msaju** [gluster/project-infrastructure#100]
Currently we have only Glusto builds run for Upstream.
Can we setup the Glusto test upstream run and get the results - on a weekly basis to start with.
We are adding lots of improvements and enhancements along with new Glusto test cases.
Upstream stability needs to be monitored on a weekly basis atleast, so that we make sure regressions are not introduced.

**Comment by mscherer** [comment]
I am a bit lost, you say we have "glusto builds run for upstream", so is it already done ? (or did you mean downstream) ? 

It seems we already had glusto run on centos CI, so this is a bit unclear to me as I lack some contexts on what happened.

Also, can you clarify who/how would you people  is the "we" in we get the results, do you have a list of people who will be responsible (and so mailed for that) ?

**Comment by msaju** [comment]
Sorry if I had confused you.

AFAIU the glusto builds for upstream is running here  - https://ci.centos.org/view/Gluster/job/gluster_glusto-patch-check/

And there is no Glusto tests run upstream like we do for downstream. Need to get it run Upstream also.

By "we", I meant the Gluster team.

**Comment by deepshikhaaa** [comment]
@msaju Yes, it is running at centos CI. Do you know if QE is already running it upstream? Because we did not set it up on downstream as well.

**Comment by msaju** [comment]
Had checked with Arthy, QE is not running on Upstream.

**Comment by mscherer** [comment]
ok so 1) what ressources are needed for that ?
2) who will take care of the job ifwhen it is broken (and so who should receive the failures that arise ? )

**Comment by msaju** [comment]
1) Below are the infrastructure details required to set up the environment for executing glusto tests.

         1. 1 driver machine for executing the tests  (could be of 4 GB RAM and 50 GB HD)
         2. 2 gluster client machines.
               - minimum configuration: 4GB RAM and 20GB HD
         3. 6 gluster server machines 
               - minimum configuration: 8GB RAM, 50GB HD, 
                                            at least 1Gbps network speed
                                            5 extra devices with 30GB each. Eg: /dev/vdb, /dev/vdc, /dev/vdd, /dev/vde, /dev/vdf
             Doc Link for setting up glusto tests - https://github.com/gluster/glusto-tests/blob/master/README.md

2) The test failures will be either genuine regressions or issues with test cases, hence failures should be informed similar to the regression failures that we send out.

**Comment by mscherer** [comment]
1.4 T of disk and ~60G of ram, at minima. What about cpus ?

**Comment by aloganat** [comment]
We require 4 vCPUs for the machines.

**Comment by deepshikhaaa** [comment]
We can make use of centos CI where all the jobs related to glusto is already running.

NOTE: Centos CI jobs needs to be migrated to the new OCP cluster. My question is who will be taking the responsibility of migrating them?
